src/app/ToDo/Components/to-do.component.ts

```diff
@@ -1,66 +1,92 @@
 import { Component, OnInit } from '@angular/core';
+import {
+  TodosState
+} from '../state'
+import * as TodoActions from '../state/todo.action'
+import { ToDo } from '../state/todo.model';
+import { Store } from '@ngrx/store';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ToDoFacade } from '../state/todo.facade';
 import { Observable, Subscription } from 'rxjs';
-import { map } from 'rxjs/operators';
-import ToDo from '../state/todo.model';
-import ToDoState from '../state/todo.state';
-import { ToDoFacade } from './../state/todo.facade';
 
 @Component({
   selector: 'app-to-do',
-  templateUrl: './to-do.component.html',
-  styleUrls: ['./to-do.component.css']
+  templateUrl: './to-do.component.html'
 })
 export class ToDoComponent implements OnInit {
-  todo$: Observable<ToDoState>;
-  ToDoSubscription: Subscription;
-  ToDoList: ToDo[] = [];
 
-  Title: string = '';
-  IsCompleted: boolean = false;
-  todoError: Error = null;
+  public todo$: Observable<any>;
+  public todoForm: FormGroup;
+  public toDoSubscription: Subscription;
 
-  constructor(private _todoFacade: ToDoFacade) {
+  public loadTask(task: ToDo) {
+    this.todoForm.patchValue(task);
+  }
+
+  constructor(
+    private _todoFacade: ToDoFacade,
+    private _store: Store<TodosState>,
+    private _formBuilder: FormBuilder) {
     this.todo$ = _todoFacade.todo$;
   }
 
-  ngOnInit() {
-    
-    //Calling the facade method, to give all tasks
-    this._todoFacade.getAllTasks();
+  public ngOnInit() {
+    this._todoFacade.getTasks();
+
+    this.buildForm();
+  }
+
+  buildForm() {
+    this.todoForm = this._formBuilder.group({
+      id: [null],
+      text: [, Validators.compose([Validators.required])],
+      completed: [false],
+    });
+  }
+
+  /**
+   * Adding or editing a task
+   */
+  public async saveTodo() {
```

Comment:
This is the first time I meet someone to use `async / await` on methods where actions are dispatched.

Can you explain to me why you used it?

Reply:
You are right, in this cases I don't need synchrony in the execution. But I use when I have some code bellow that need some function execution finished to continue.